File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/newsequence/justorempty/JustOrEmptyEx3.java
package boki.learnjava.newsequence.justorempty;

import boki.learnjava.utils.Logger;
import reactor.core.publisher.Mono;

import java.util.Optional;

/**
 * justOrEmpty() 사용 예
 * 인자로 Optional.isPresent()가 true가 아니라면, onNext emit 없이 onComplete만 emit
 */
public class JustOrEmptyEx3 {
    public static void main(String[] args) {
        Mono
            .justOrEmpty(Optional.ofNullable(null))
            .log()
            .subscribe(Logger::onNext);
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v1_newsequence/a_justorempty/JustOrEmptyEx3.kt
package boki.learnkt.v1_newsequence.a_justorempty

import boki.learnkt.util.Logger
import reactor.core.publisher.Mono
import java.util.*

/**
 * justOrEmpty() 사용 예
 * 인자로 Optional.isPresent()가 true가 아니라면, onNext emit 없이 onComplete만 emit
 */
fun main() {
    Mono.justOrEmpty(Optional.ofNullable(null))
        .log()
        .subscribe { Logger.onNext(it) }
}